xilem/src/one_of.rs

```diff
@@ -0,0 +1,373 @@
+        elem: xilem_core::one_of::OneOf9<
+            Pod<A>,
+            Pod<B>,
+            Pod<C>,
+            Pod<D>,
+            Pod<E>,
+            Pod<F>,
+            Pod<G>,
+            Pod<H>,
+            Pod<I>,
+        >,
+    ) -> Self::OneOfElement {
+        match elem {
+            xilem_core::one_of::OneOf9::A(w) => Pod::new(OneOfWidget::A(w.inner)),
+            xilem_core::one_of::OneOf9::B(w) => Pod::new(OneOfWidget::B(w.inner)),
+            xilem_core::one_of::OneOf9::C(w) => Pod::new(OneOfWidget::C(w.inner)),
+            xilem_core::one_of::OneOf9::D(w) => Pod::new(OneOfWidget::D(w.inner)),
+            xilem_core::one_of::OneOf9::E(w) => Pod::new(OneOfWidget::E(w.inner)),
+            xilem_core::one_of::OneOf9::F(w) => Pod::new(OneOfWidget::F(w.inner)),
+            xilem_core::one_of::OneOf9::G(w) => Pod::new(OneOfWidget::G(w.inner)),
+            xilem_core::one_of::OneOf9::H(w) => Pod::new(OneOfWidget::H(w.inner)),
+            xilem_core::one_of::OneOf9::I(w) => Pod::new(OneOfWidget::I(w.inner)),
+        }
+    }
+
+    fn update_one_of_element_mut(
+        elem_mut: &mut xilem_core::Mut<'_, Self::OneOfElement>,
+        new_elem: xilem_core::one_of::OneOf9<
+            Pod<A>,
+            Pod<B>,
+            Pod<C>,
+            Pod<D>,
+            Pod<E>,
+            Pod<F>,
+            Pod<G>,
+            Pod<H>,
+            Pod<I>,
+        >,
+    ) {
+        let new_inner = match new_elem {
+            xilem_core::one_of::OneOf9::A(w) => OneOfWidget::A(w.inner),
+            xilem_core::one_of::OneOf9::B(w) => OneOfWidget::B(w.inner),
+            xilem_core::one_of::OneOf9::C(w) => OneOfWidget::C(w.inner),
+            xilem_core::one_of::OneOf9::D(w) => OneOfWidget::D(w.inner),
+            xilem_core::one_of::OneOf9::E(w) => OneOfWidget::E(w.inner),
+            xilem_core::one_of::OneOf9::F(w) => OneOfWidget::F(w.inner),
+            xilem_core::one_of::OneOf9::G(w) => OneOfWidget::G(w.inner),
+            xilem_core::one_of::OneOf9::H(w) => OneOfWidget::H(w.inner),
+            xilem_core::one_of::OneOf9::I(w) => OneOfWidget::I(w.inner),
+        };
+        let old_inner = std::mem::replace(elem_mut.widget, new_inner);
+        match old_inner {
+            OneOfWidget::A(w) => elem_mut.ctx.remove_child(w),
+            OneOfWidget::B(w) => elem_mut.ctx.remove_child(w),
+            OneOfWidget::C(w) => elem_mut.ctx.remove_child(w),
+            OneOfWidget::D(w) => elem_mut.ctx.remove_child(w),
+            OneOfWidget::E(w) => elem_mut.ctx.remove_child(w),
+            OneOfWidget::F(w) => elem_mut.ctx.remove_child(w),
+            OneOfWidget::G(w) => elem_mut.ctx.remove_child(w),
+            OneOfWidget::H(w) => elem_mut.ctx.remove_child(w),
+            OneOfWidget::I(w) => elem_mut.ctx.remove_child(w),
+        }
+        elem_mut.ctx.children_changed();
+    }
+}
+
+impl crate::core::one_of::PhantomElementCtx for ViewCtx {
+    type PhantomElement = Pod<Box<dyn Widget>>;
```

Comment:
I guess when we would do something similar as in xilem_web (like `WithStyle`, `WithClasses` etc.) we would need some `PhantomWidget` which implements all kinds of these traits, similar as `Noop` in xilem_web.

Reply:
Yes, indeed. That would be unneeded complexity as-is, though.

Reply:
> That would be unneeded complexity as-is

You mean something like `Noop: Widget`, since `Box<dyn Widget>` already implements `Widget`?

Reply:
Yeah I agree. That should only be done, when it really makes sense.